Client/Frontend/Sync/SyncAddDeviceViewController.swift

```diff
@@ -98,38 +98,49 @@ class SyncAddDeviceViewController: SyncViewController {
         containerView.layer.masksToBounds = true
 
         guard let syncSeed = Sync.shared.syncSeedArray else {
-            // TODO: Pop and error
+            showInitializationError()
             return
         }
+        
+        let syncCrypto = SyncCrypto()
 
-        let qrSyncSeed = SyncCrypto.shared.joinBytes(fromCombinedBytes: syncSeed)
+        let qrSyncSeed = syncCrypto.joinBytes(fromCombinedBytes: syncSeed)
         if qrSyncSeed.isEmpty {
-            // Error
+            showInitializationError()
             return
         }
-
-        SyncCrypto.shared.passphrase(fromBytes: syncSeed) { (words, error) in
-            guard let words = words, error == nil else {
-                return
-            }
-            
-            self.qrCodeView = SyncQRCodeView(data: qrSyncSeed)
-            self.containerView.addSubview(self.qrCodeView!)
-            
-            self.qrCodeView!.snp.makeConstraints { (make) in
+        
+        let words = syncCrypto.passphrase(fromBytes: syncSeed)
+        
+        switch words {
+        case .success(let passphrase):
+            qrCodeView = SyncQRCodeView(data: qrSyncSeed)
+            containerView.addSubview(qrCodeView!)
+            qrCodeView?.snp.makeConstraints { make in
                 make.top.bottom.equalTo(0).inset(22)
                 make.centerX.equalTo(self.containerView)
                 make.size.equalTo(BarcodeSize)
             }
             
-            self.codewordsView.text = words.joined(separator: " ")
-            
-            self.SEL_changeMode()
+            self.codewordsView.text = passphrase.joined(separator: " ")
+        case .failure:
+            showInitializationError()
+            return
         }
         
         self.setupVisuals()
     }
     
+    private func showInitializationError() {
```

Comment:
This looks mostly copied from the welcome screen, correct? If so we need to abstract this somewhere too.

Reply:
I will add it here https://github.com/brave/brave-ios/pull/991, there's a new structure that groups Sync related alerts. Please merge this PR first :)